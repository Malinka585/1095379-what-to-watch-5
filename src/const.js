export const GENRE_DEFAULT = `All genres`;
export const FILMS_CONT_PER_STEP = 8;
export const TabsHeaders = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

