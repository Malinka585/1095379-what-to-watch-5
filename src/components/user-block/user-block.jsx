import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {AuthorizationStatus} from "../../const";
import {Link} from "react-router-dom";

const UserBlock = (props) => {
  const {authorizationStatus} = props;

  return (
    <div className="user-block">
      {authorizationStatus === AuthorizationStatus.AUTH ?
        <Link to={`/mylist`}>
          <div className="user-block__avatar">
            <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
          </div>
        </Link>
        :
        <Link to={`/login`} className="user-block__link">Sign in</Link>}
    </div>
  );
};

const mapStateToProps = ({USER}) => ({
  authorizationStatus: USER.authorizationStatus,
});

UserBlock.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
};

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
