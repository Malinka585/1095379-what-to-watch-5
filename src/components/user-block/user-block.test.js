import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from "react-router-dom";
import {AuthorizationStatus} from "../../const";

import {UserBlock} from "./user-block";

const {AUTH, NO_AUTH} = AuthorizationStatus;


describe(`Should UserBlock render correctly`, () => {
  it(`AuthorizationStatus auth`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock
              authorizationStatus={AUTH}
            />
          </BrowserRouter>

      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`AuthorizationStatus no-auth`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <UserBlock
              authorizationStatus={NO_AUTH}
            />
          </BrowserRouter>

      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
