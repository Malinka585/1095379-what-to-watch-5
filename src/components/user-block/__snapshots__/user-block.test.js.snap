// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should UserBlock render correctly AuthorizationStatus auth 1`] = `
<div
  className="user-block"
>
  <div
    className="user-block__avatar"
  >
    <img
      alt="User avatar"
      height="63"
      src="img/avatar.jpg"
      width="63"
    />
  </div>
</div>
`;

exports[`Should UserBlock render correctly AuthorizationStatus no-auth 1`] = `
<div
  className="user-block"
>
  <a
    className="user-block__link"
    href="/login"
    onClick={[Function]}
  >
    Sign in
  </a>
</div>
`;
